torch/distributed/checkpoint/storage.py

```diff
@@ -19,6 +20,39 @@ class WriteResult:
     storage_data: Any
 
 
+@runtime_checkable
+class AsyncStager(Protocol):
+    """
+    AsyncStager protocol for objects that can be staged asynchronously. This protocol is meant
+    to provide extensibility for dcp.async_save, allowing users to customize how data is staged
+    previous to executing the usual dcp.save path in parallel.
+    """
+
+    # default to True since most people will stage synchronously
+    _synchronize_after_execute: bool = True
+
+    @property
```

Comment:
Is this a property just to have a docstring?
if so, I think you can just have this docstring below the attribute according to https://peps.python.org/pep-0257/

Reply:
I think this was actually to force the `protocol` aspect, which I don't think works for attributes. I'll double check